chords.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+import speed
+
+
+def get_chords(df, threshold=0.05):
+    pressed_simultaneously_df = speed.get_notes_idxs_pressed_simultaneously(
+        df=df, threshold=threshold
+    )
+
+    records = []
+    for it, row in pressed_simultaneously_df.iterrows():
+        idxs = row.pressed_simultaneously
+        ends = df.iloc[idxs].end.values
+        where_close_to_min = np.where(ends - min(ends) < threshold)[0]
+        close_end_idxs = np.array(idxs)[where_close_to_min]
+        if len(close_end_idxs) > 1:
+            notes_idxs = list(close_end_idxs)
+            record = {
+                "notes_idxs": notes_idxs,
+                "time": df.iloc[notes_idxs].start.values.mean(),
+                "pitches": list(df.iloc[notes_idxs].pitch.values),
```

Comment:
```suggestion
                "pitches": df.iloc[notes_idxs].pitch.tolist(),
```

`pd.Series` already has a list conversion :)

Reply:
I fixed the chords.py file implementing your suggestion and adding another part - table with number of occurrences of chords:
![Screenshot from 2023-07-24 14-00-01](https://github.com/Nospoko/midi-internship/assets/117731792/9811a749-eeb5-425a-8f1c-c678e51103c2)
Moreover, I created the validation challenge you mentioned about above and the result is the plot with repeated chords in time: 
![image](https://github.com/Nospoko/midi-internship/assets/117731792/cdbd1927-a690-4491-8f7b-d259ddff3328)

Reply:
A chord is often defined as three distinct notes, but in your table top rows are all just pairs of notes. Moreover, looks like all of them are 12 tones apart, which makes them octaves - so technically these are a single note (60 is C4 and 72 is C5, but both are C).

To find chords consisting of three or more different notes you'd need too add some filtering to your dataframe.